src/module/item/entity.js

```diff
@@ -84,29 +84,33 @@ export class OseItem extends Item {
     };
 
     if (itemData.missile && itemData.melee && !isNPC) {
-      // Dialog
-      new Dialog({
-        title: "Choose Attack Range",
-        content: "",
-        buttons: {
-          melee: {
-            icon: '<i class="fas fa-fist-raised"></i>',
-            label: "Melee",
-            callback: () => {
-              this.actor.targetAttack(rollData, "melee", options);
+      if (options.preferedType === "melee" || options.preferedType === "missile"){
+        type = options.preferedType;
+      } else {
+        // Dialog
+        new Dialog({
+          title: "Choose Attack Range",
```

Comment:
While you're in here, could you replace strings like this with a localized string?

Reply:
done